tools/ci_build/build.py

```diff
@@ -687,6 +687,7 @@ def convert_arg_line_to_args(self, arg_line):
     parser.add_argument("--use_cache", action="store_true", help="Use compiler cache in CI")
 
     parser.add_argument("--use_triton_kernel", action="store_true", help="Use triton compiled kernels")
+    parser.add_argument("--use_lock_free_queue", action="store_true", help="Use lock-free task queue for threadpool.")
```

Comment:
Can you add some documentation with regards to why this option is needed? This will help users decide if they should use this or not. Also, is this mentioned in the performance page somewhere or different PR?

Reply:
Plan to explain this in the documentation branch.
For here - wouldn't "Use lock-free task queue for threadpool." be clear enough?